code/script.js

```diff
@@ -29,23 +40,247 @@ const showMessage = (message, sender) => {
+      setTimeout(botChatEnd, 1000);
+    } else if (q6Buttons.target.value === "no") {
+      let userConfirmation = "no"
+      showMessage(`${userConfirmation}`, 'user');
+      showMessage(`You're request has been cancelled`, 'bot');
+      showMessage(`We hope you visit again soon, Good bye`, 'bot');
+      setTimeout(botChatEnd, 1000);
+    }
+  })
+}
+
+
+//---------------- Bot Chat End -------------------------//
+
+const botChatEnd = () => {
+  console.log("You have entered the end function"); //reference for me to help me track the flow
+
+  //remove buttons and re-create the input field so it looks the same as beginning
+  const areaOptions = document.getElementById("input-wrapper")
+  areaOptions.innerHTML = `
+    <form id="name-form">
+    <input id="name-input" type="text" placeholder="Type here...." autofocus="autofocus" />
+    <button class="send-btn" type="submit">
+      Send
+    </button>
+  </form>
+  `
+  //site will reload when user types any input into the text and presses the submit button
+}
+
+// Set up your eventlisteners here - I have ONLY ONE event listener OUTSIDE my functions. The following is suggested in the instructions.md...I could use it to validate the first 2 questions (that use the input text bar) but after that I have eventlisteners in each function to handle events from buttons, date, and dropdown menu. I don't think this is neat code, but it works.
+
+//---------------- Event Listener for Input text, Question 1 and 2 only -------------------------//
+
+logSubmit = (event) => {  //listen for when form is submitted (submit button or enter is pressed)
+  console.log(`Question count inside logsubmit:`, questionCount) //reference for me to help me track the flow, I had issues with this
+
+  event.preventDefault(); //still trying to understand how this works
+  const nameSubmitted = document.getElementById("name-input")
+  usersName = nameSubmitted.value;
+  showMessage(`${usersName}`, 'user');
+  nameSubmitted.value = "";
+  if (questionCount == 0) {
+    questionCount = 1;
+    setTimeout(botQuestion2, 1000); //I had problems with where to put this, got help from a code snippet in our Slack :-)
+  } else if (questionCount == 1) {
+    questionCount = 2;
+    setTimeout(botQuestion3, 1000);
+
+    //----after Question 3 this function is no longer used, I have added eventlisteners INSIDE the functons for further questions as I could not work out how to call this one again when using buttons, date, or select dropdown as inout. I realise this isn't a good flow (Q1 and Q2 share their eventlistener which is outside their function) but this is how I got it to work :-)-----//
+  }
+}
+
+//when form is submitted
+const form = document.getElementById("name-form");
+form.addEventListener("submit", logSubmit);
+
+//----- First function - When the website is loaded, the chatbot asks first question, after a 1 second delay ------//
+
+setTimeout(botQuestion1, 1000) //Step 1 - played around with the value
```

Comment:
I moved this one up to be underneath the message - unclear to me if there is a reason to have it in the bottom as the instructions said. 

Also for event listeners, I had all of them within functions - not sure why they should be outside? I understood the instrutions differently - instead of having all functions in the top, event listeners in middle, etc i understood it as the order to build up a function:
write your function 
event listener for funcion
set timeout for next function
write your next function

I now see how we understood it differently, and I am not sure which way is the "best" or "cleanest". I hope you understand what I mean!